srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java

```diff
@@ -74,6 +77,19 @@ class CatalogServiceHandler implements EventHandler {
 		this.analyzer = CqnAnalyzer.create(model);
 	}
 
+	@Before(entity = Books_.CDS_NAME)
+	public void beforeReadBooks(CdsReadEventContext context) {
+		CqnSelect copy = CQL.copy(context.getCqn(), new Modifier() {
+			@Override
+			public List<CqnSelectListItem> items(List<CqnSelectListItem> items) {
+				items.add(CQL.get("details"));
+				items.add(CQL.get("stock"));
```

Comment:
What if those are already selected? I think this logic simply puts this onto the select list twice or is there some automatic deduplication?

Reply:
Good question, I tried it with List.contains, but this doesn't seem to work. It adds the same item twice. Looks like there is no sufficient implementation of equals for the CqnSelectListItem implementations.
Not sure what is the best practice here.

Reply:
ah, forget my last comment, it works with List.contains. Stupid error on my side using VSCode without auto.build enabled :-).